mypy/fscache.py

```diff
@@ -40,37 +41,16 @@ def __init__(self) -> None:
 
     def flush(self) -> None:
         """Start another transaction and empty all caches."""
-        self.stat_cache = {}  # type: Dict[str, os.stat_result]
-        self.stat_error_cache = {}  # type: Dict[str, Exception]
-        self.listdir_cache = {}  # type: Dict[str, List[str]]
-        self.listdir_error_cache = {}  # type: Dict[str, Exception]
```

Comment:
Note that [`lru_cache` does not support this feature](https://bugs.python.org/issue21537)

Reply:
Yes I will be adding an exception cache next. I'm not entirely sure caching exceptions is a good idea (since this could lead to difficulties debugging), but I should probably leave as is.

Reply:
Caching exceptions is needed for transactional semantics. If we don't cache them, the second call targeting the same path could succeed. (Also, performance could be impacted if we perform the same failing operation many times, but this is perhaps unlikely.)